server/controllers/finance/cash.js

```diff
@@ -253,32 +235,11 @@ function update(req, res, next) {
 }
 
 /**
- * GET /cash/references/:reference
- * retrieves cash payment uuids from a reference string (e.g. HBB123)
- */
-function reference(req, res, next) {
-  // alias the reference
-  var ref = req.params.reference;
-
-  const sql = `
-    SELECT BUID(c.uuid) AS uuid FROM (
-      SELECT cash.uuid
-      FROM cash JOIN project ON cash.project_id = project.id
-    )c WHERE c.reference = ?;
-  `;
-
-  db.one(sql, [ref], ref, 'cash')
-    .then((payment) => {
-      // references should be unique - return the first one
-      res.status(200).json(payment);
-    })
-    .catch(next)
-    .done();
-}
-
-/**
  * GET /cash/:checkin/:invoiceUuid
  * Check if the invoice is paid
+ * TODO(@jniles) - this should use a more intelligent system to see if an
```

Comment:
What will happen if the invoice is not paid but posted?

Reply:
It will be rejected by the `/transactions/:uuid` route.  Specifically, this line: https://github.com/IMA-WorldHealth/bhima-2.X/pull/2188/files#diff-3a4662a4da5c4801c73af27c7f7358ddR131

Reply:
Ha ha, it was strange because I saw nothing at this link also